src/foam/nanos/approval/ApprovableAwareDAO.js

```diff
@@ -165,7 +206,7 @@ foam.CLASS({
       ApprovableAware approvableAwareObj = (ApprovableAware) obj;
       LifecycleAware lifecycleObj = (LifecycleAware) obj;
 
-      DAO approvalRequestDAO = (DAO) x.get("approvalRequestDAO");
+      DAO approvalRequestDAO = obtainApprovalRequestDAO(x, obj);
```

Comment:
This doesn't appear to be used.

Reply:
getApprovalRequestDAO() should setup the dao.  The javaFactory of 'approvalRequestDAO' can have the logic of obtainApprovalRequestDAO - using getX(). 

The only other alternative is to make this DAO  implement foam.nanos.NanoService, implement 'start',  and then aquire the approvalRequestDAO.

Reply:
We want the relationship DAO here and it needs the object `obj` which can be a different object on each put.

`obtainApprovalRequestDAO` is the first call to resolve the approval request relationship DAO if it exists, subsequently we can either use obtainApprovalRequestDAO or getApprovalRequest.

Reply:
Updated to resolveApprovalRequestDAO which will resolve the relationship DAO and set it as the approvalRequestDAO and added javaFactory on approvalRequestDAO property - https://github.com/foam-framework/foam2/pull/3580/commits/2a4a1e9db3dd9f8c1758a06079ae7f1f3f10bf61.